tools/swift-inspect/Sources/swift-inspect/DumpConcurrency.swift

```diff
@@ -0,0 +1,401 @@
+    metadataIsActorCache[metadata] = result
+    return result
+  }
+
+  func name(metadata: swift_reflection_ptr_t) -> String? {
+    if let cached = metadataNameCache[metadata] {
+      return cached
+    }
+
+    let name = context.name(metadata: metadata)
+    metadataNameCache[metadata] = name
+    return name
+  }
+
+  func info(forTask task: swift_reflection_ptr_t) -> TaskInfo? {
+    let reflectionInfo = swift_reflection_asyncTaskInfo(context, task)
+    if let error = reflectionInfo.Error {
+      print("Error getting info for async task \(hex: task): \(String(cString: error))")
+      return nil
+    }
+
+    // These arrays are temporary pointers which we must copy out before we call
+    // into Remote Mirror again.
+    let children = Array(UnsafeBufferPointer(
+        start: reflectionInfo.ChildTasks,
+        count: Int(reflectionInfo.ChildTaskCount)))
+    let asyncBacktraceFrames = Array(UnsafeBufferPointer(
+        start: reflectionInfo.AsyncBacktraceFrames,
+        count: Int(reflectionInfo.AsyncBacktraceFramesCount)))
+
+    var allocatorSlab = reflectionInfo.AllocatorSlabPtr
+    var allocatorTotalSize = 0
+    var allocatorTotalChunks = 0
+    while allocatorSlab != 0 {
+      let allocations = swift_reflection_asyncTaskSlabAllocations(context,
+                                                                  allocatorSlab)
+      guard allocations.Error == nil else { break }
+      allocatorTotalSize += Int(allocations.SlabSize)
+      allocatorTotalChunks += Int(allocations.ChunkCount)
+
+      allocatorSlab = allocations.NextSlab
+    }
+
+    return TaskInfo(
+      address: task,
+      jobFlags: reflectionInfo.JobFlags,
+      taskStatusFlags: reflectionInfo.TaskStatusFlags,
+      id: reflectionInfo.Id,
+      runJob: reflectionInfo.RunJob,
+      allocatorSlabPtr: reflectionInfo.AllocatorSlabPtr,
+      allocatorTotalSize: allocatorTotalSize,
+      allocatorTotalChunks: allocatorTotalChunks,
+      childTasks: children,
+      asyncBacktrace: asyncBacktraceFrames
+    )
+  }
+
+  func taskHierarchy() -> [(level: Int, lastChild: Bool, task: TaskInfo)] {
+    var hierarchy: [(level: Int, lastChild: Bool, task: TaskInfo)] = []
+
+    let topLevelTasks = tasks.values.filter{ $0.parent == nil }
+    for top in topLevelTasks.sorted(by: { $0.id < $1.id }) {
+      var stack: [(index: Int, task: TaskInfo)] = [(0, top)]
+      hierarchy.append((0, true, top))
+
+      while let (index, task) = stack.popLast() {
+        if index < task.childTasks.count {
+          stack.append((index + 1, task))
+          let childPtr = task.childTasks[index]
+          let childTask = tasks[childPtr]!
+          hierarchy.append((stack.count, index == task.childTasks.count - 1, childTask))
+          stack.append((0, childTask))
+        }
+      }
+    }
+    return hierarchy
+  }
+
+  func remove(from: String, upTo: String) -> String {
+    from.withCString {
```

Comment:
Trying to avoid importing Foundation, on vague concerns that it will make it more difficult to build, especially if/when we port this tool to other platforms.